Documentation/ApiOverview/Hooks/Events/Extbase/Persistence/EntityAddedToPersistenceEvent.rst

```diff
@@ -6,8 +6,9 @@
 EntityAddedToPersistenceEvent
 =============================
 
-Event which is fired after an object/entity was sent to persistence layer to be added,
-but before updating the reference index and current session.
+Event which is fired after an object/entity was persisted on add.
+
+The event is dispatched before updating the reference index, after adding the object to the persistence session.
```

Comment:
The last part is wrong nowe however. There are four (+1) steps:

1. [Persist the object](https://github.com/TYPO3/typo3/blob/2f940fd527acfc719fbfdbe8a9f789ec18ba009a/typo3/sysext/extbase/Classes/Persistence/Generic/Backend.php#L632)
2. **[Dispatch the `EntityAddedToPersistenceEvent` event](https://github.com/TYPO3/typo3/blob/2f940fd527acfc719fbfdbe8a9f789ec18ba009a/typo3/sysext/extbase/Classes/Persistence/Generic/Backend.php#L636)**
3. [Update the reference index](https://github.com/TYPO3/typo3/blob/2f940fd527acfc719fbfdbe8a9f789ec18ba009a/typo3/sysext/extbase/Classes/Persistence/Generic/Backend.php#L640)
4. [Add the object to the persistence session](https://github.com/TYPO3/typo3/blob/2f940fd527acfc719fbfdbe8a9f789ec18ba009a/typo3/sysext/extbase/Classes/Persistence/Generic/Backend.php#L642)
5. ([Dispatch the undocumented `EntityFinalizedAfterPersistenceEvent` event](https://github.com/TYPO3/typo3/blob/2f940fd527acfc719fbfdbe8a9f789ec18ba009a/typo3/sysext/extbase/Classes/Persistence/Generic/Backend.php#L644))

So we need to clarify the position of the 2nd step. Before it sounded like the `EntityAddedToPersistenceEvent` is dispatched as 1st step.

Reply:
I'll rewrite it with your infos. I think the main problem I had was the difference between "persisting" and adding it to the persistence session.